index.js

```diff
@@ -170,10 +170,12 @@ Object.defineProperty(Primus.prototype, 'client', {
 //
 Object.defineProperty(Primus.prototype, 'Socket', {
   get: function () {
-    return require('load').compiler(this.library(true), 'primus.js', {
-      __filename: 'primus.js',
-      __dirname: process.cwd()
-    }).Primus;
+    var compiler = new module.constructor();
```

Comment:
just wondering if it doesn't make more sense to do `var Module = require('module')` here instead of assuming that module is a constructor.